### PR TITLE
Fixes in corehost for building x86 arch.

### PR DESCRIPTION
**Pull request by Sridhar-MS**
- Set the calling convention to '__stdcall' explicitly else the default is '_cdecl' on windows which mismatches with coreclr hosting APIs.
- Use the newly built 'corehost' in stage1 instead of one from stage0.

cc @schellap

### CODE REVIEW

File: src/corehost/cli/coreclr.cpp [hunk 1]
```diff
@@ -9,7 +9,7 @@
 static pal::dll_t g_coreclr = nullptr;
 
 // Prototype of the coreclr_initialize function from coreclr.dll
-typedef pal::hresult_t(*coreclr_initialize_fn)(
+typedef pal::hresult_t(__stdcall *coreclr_initialize_fn)(
```
Comment (schellap) [inline src/corehost/cli/coreclr.cpp]: Compiles on unix?

**Comment by piotrpMSFT** [comment]
@dotnet-bot test OSX Debug Build please.

**Comment by piotrpMSFT** [comment]
once @schellap's comment is handled, :shipit:

File: scripts/dotnet-cli-build/CompileTargets.cs [hunk 2]
```diff
@@ -195,6 +195,10 @@ private static BuildTargetResult CompileStage(BuildTargetContext c, DotNetCli do
                 }
             }
 
+            // dotnet.exe is from stage0. But we must be using the newly built corehost in stage1
```
Comment (piotrpMSFT) [inline scripts/dotnet-cli-build/CompileTargets.cs]: This is bootstrapping code, I assume...

Comment (Sridhar-MS) [inline scripts/dotnet-cli-build/CompileTargets.cs]: Well stage1 dotnet.exe is not the locally built corehost. Instead it is copied from stage0 (due to `dotnet publish`). So any fix in corehost src code cannot be bootstrapped in our builds easily. So I am planning to leave it as is.

File: src/corehost/cli/coreclr.cpp (hunk 1 above)
Comment (Sridhar-MS) [inline src/corehost/cli/coreclr.cpp]: Fixed

**Comment by schellap** [comment]
LGTM. Thanks.

**Pull request merged**